pkg/apis/extensions/validation/validation.go

```diff
@@ -152,6 +165,8 @@ func ValidateThirdPartyResource(obj *extensions.ThirdPartyResource) field.ErrorL
 		version := &obj.Versions[ix]
 		if len(version.Name) == 0 {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("versions").Index(ix).Child("name"), version, "must not be empty"))
+		} else if validation.IsDNS1123Label(version.Name) {
```

Comment:
@lavalamp @caesarxuchao @deads2k @kubernetes/sig-api-machinery opinions on allowed format for an api version?